formats/json/commonMain/src/kotlinx/serialization/json/JsonAnnotations.kt

```diff
@@ -74,3 +74,36 @@ public annotation class JsonNames(vararg val names: String)
 @Target(AnnotationTarget.CLASS)
 @ExperimentalSerializationApi
 public annotation class JsonClassDiscriminator(val discriminator: String)
+
+
+/**
+ * Specifies whether encounters of unknown properties (i.e., properties not declared in the class) in the input JSON
+ * should be ignored instead of throwing [SerializationException].
+ *
+ * With this annotation, it is possible to allow unknown properties for specific classes, while
```

Comment:
Annotation's effect does not propagate to annotated class' fields:

```kotlin
@JsonIgnoreUnknownKeys @Serializable
class Outer(val inner: Inner)

@Serializable
class Inner(val x: Int)

...
// {"untitled":"unknown", "inner": {"x": 42}} // OK
// {"inner": {"x": 42, "untitled": "unknown"}} // FAIL
```

Was it a deliberate choice?
Personally, I think the current behavior is fine, but it might be worth to stress that `for specific classes` means that the annotation has a shallow effect.

Reply:
Yes, it is a deliberate choice (also we don't have technical means for this). 
I'll change 'specific' to 'annotated'.